crates/astria-composer/src/searcher/executor.rs

```diff
@@ -164,28 +216,43 @@ impl Executor {
                             break Err(e).wrap_err("failed submitting bundle to sequencer");
                         }
                     }
+
+                    block_timer.as_mut().reset(Instant::now() + Duration::from_millis(self.block_time_ms));
                 }
 
-                // receive new bundle for processing
-                Some(bundle) = self.new_bundles.recv(), if submission_fut.is_terminated() => {
-                    // TODO(https://github.com/astriaorg/astria/issues/476): Attach the hash of the
-                    // bundle to the span. Linked GH issue is for agreeing on a hash for `SignedTransaction`,
-                    // but both should be addressed.
-                    let span =  info_span!(
-                        "submit bundle",
-                        nonce.initial = nonce,
-                        bundle.len = bundle.len(),
+                Some(bundle) = bundle_factory.pop_complete(), if submission_fut.is_terminated() => {
+                    debug!(
```

Comment:
I agree - this should be replaced with properly managing spans: create one on tx collection, then merge them into a bundle span or something like that. Keeping this in here until that's all added seems fine to me tho? I'll change it to trace level so that it's not as noisy.

Reply:
I commented on this in `fn make_submission_fut` (which should probably be named "submit_bundle"): if you properly instrument it you a) don't need to create an `info_span!` inside it, and b) you can put this `debug!` event there so that it will be emitted under the correct span.